src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariantsIntegrationTest.java

```diff
@@ -343,6 +343,54 @@ public void testNoGTs() throws IOException {
         spec.executeTest("testNoGTs--" + testFile, this);
     }
 
+    @Test
+    public void testRemoveMonomorphAfterSNSelect() throws IOException {
+        final String testFile = getToolTestDataDir() + "spanning_deletion.vcf";
+        final String samplesFile = "NA1";
+
+        final IntegrationTestSpec spec = new IntegrationTestSpec(
+                baseTestString(" -sn " + samplesFile + " --remove-unused-alternates --exclude-non-variants", testFile),
+                Collections.singletonList(getToolTestDataDir() + "expected/" + "testSelectVariants_RemoveMonomorphAfterSNSelect.vcf")
+        );
+        spec.executeTest("test will make sure line is removed when only monomorphic allele exists" + testFile, this);
+    }
+
+    @Test
+    public void testRemoveSingleSpanDelAlleleNoSpanDel() throws IOException {
+        final String testFile = getToolTestDataDir() + "spanning_deletion.vcf";
+        final String samplesFile = "NA1";
+
+        final IntegrationTestSpec spec = new IntegrationTestSpec(
+                baseTestString(" -sn " + samplesFile + " --remove-unused-alternates --exclude-non-variants", testFile),
+                Collections.singletonList(getToolTestDataDir() + "expected/" + "testSelectVariants_RemoveSingleSpanDelAlleleNoSpanDel.vcf")
+        );
+        spec.executeTest("test will not remove variant line where '*' is only ALT allele because none exist" + testFile, this);
```

Comment:
I don't see how this is different from the above test.

Reply:
Did you want to use a different input file where the * is in a heterozygous genotype?